controllers/front/CartController.php

```diff
@@ -69,6 +71,21 @@ public function init()
         $this->qty = abs(Tools::getValue('qty', 1));
         $this->id_address_delivery = (int)Tools::getValue('id_address_delivery');
         $this->preview = ('1' === Tools::getValue('preview'));
+
+        /** Check if the products in the cart are available */
+        $product = $this->context->cart->checkQuantities(true);
+
+        if (Tools::getIsset('action') && "show" === Tools::getValue('action')) {
+            if (is_array($product)) {
```

Comment:
This test is not enough. `checkQuantities()` could also return `false` which would also mean that something is wrong.
I suggest checking if `$product !== true`. Then, when you need to build the error message, testing if `$product` is an array to build the message accordingly.